src/recipe_dictionary.h

```diff
@@ -0,0 +1,70 @@
+#ifndef RECIPE_DICTIONARY_H
+#define RECIPE_DICTIONARY_H
+
+#include "crafting.h" // for recipe, FIXME: circular dependency -> possible fix: move recipe to this file?
+
+#include <string>
+#include <vector>
+#include <map>
+#include <list>
+
+struct recipe;
+
+/**
+*   Repository class for recipes.
+*
+*   This class is aimed at making (fast) recipe lookups easier from the outside.
+*/
+class recipe_dictionary {
+    public:
+        void add( recipe *rec );
+        void remove( recipe *rec );
+        void clear();
+
+        /** Returns a list of recipes in the 'cat' category */
+        const std::vector<recipe *>& in_category( const std::string &cat );
```

Comment:
Style nit, the reference operator '&' is placed against the label 'in_category' instead of with the return type specification.